### DISCUSSION TITLE
Decide on a New Name

### DISCUSSION THREAD

**Discussion by dmweir** [pitt-rnel/pylsb#27] [Ideas]

**Comment by jmw182** [comment]
My ideas:
- hermes messaging
- iris messaging

There are other software projects using both of these names (hermes and iris), but we could avoid name conflicts by appending with "_msg"

**Comment by jmw182** [comment]
- pigeon
As in, a carrier pigeon

**Comment by jmw182** [comment]
- arcus

This is the Roman name for Iris. There are some other projects with this name too though.

**Comment by jmw182** [comment]
- malach

Probably getting too religious (and hard to pronounce) here, but this means "messenger" in Hebrew, and is loosely both "messenger" and "angel" in the Torah.


Hermes is still my favorite name of all of these.
![giphy](https://media.giphy.com/media/l4pSZgIpG6CqxEqmQ/giphy.gif)

**Comment by twsimpso** [comment]
A couple of words that came to mind:
Lab, Experiment, Core, Data, Message, Event, Packet, Stream, Router, Dispatch, Relay, Queue, Flow, Layer, Service, Manager

**Comment by jmw182** [comment]
Some words to add to that: Simple, Networking, Architecture, Application, Client, Broker, Interface, Connection, Framework, Centralized, Distributed, Modular, Node, Carrier, Courier, Delivery, Transmit, Traffic, Handler, ...

**Comment by jmw182** [comment]
SANTA
Simple API for Network Transmission Architecture

Also, supposedly Santa delivers packages. I don't know for sure because he's never shown up for me (he must use UDP instead of TCP).

**Comment by dmweir** [comment]
I kind of lean towards a straight forward name because it makes things more obvious in code and helps people find things if searching for it when shared publicly. I think what could differentiate this is that we could tailor it to our lab use case versus other similar things that were designed for web applications.

I kind of like some sort of utility service (phone, electric, radio, tv) analogy.

Some other words to add: Hub, Broadcast, Central, Traffic Controller

Lab Packet Dispatcher/Router
Lab Data Router
Lab SwitchBoard
Experiment Messaging System
Simple Messaging System

**Comment by jmw182** [comment]
REPLY: Rnel Experiment Packet LaYer

**Comment by jmw182** [comment]
SNIFF: Simple Node-based Information Flow Framework

**Comment by jmw182** [comment]
I think Lab SwitchBoard is my favorite of @dmw109's suggestions above. The last two share acronyms with emergency medical service and sms text messages.

**Comment by jmw182** [comment]
Here are the words we've come up with in alphabetical order (for ease of making backronyms):

Application
Architecture
Broadcast
Broker
Carrier
Centralized
Channel
Client
Connection
Controller
Core
Courier
Data
Delivery
Dispatch
Distributed
Event
Experiment
Flow
Framework
Handler
Hub
Information
Interface
Lab
Layer
Manager
Message
Modular
Networking
Node
Packet
Parcel
Queue
Relay
RNEL
Router
Service
Simple
Stream
Switchboard
Traffic
Transmit
Transport

**Comment by twsimpso** [comment]
I've been leaning towards Lab SwitchBoard since last time we discussed options.  I think the name holds up well - it is descriptive and memorable.  Still welcome input.

**Comment by jmw182** [comment]
Renamed pyrtma repo to pylsb (lab switchboard).
